internal/validations/clusterlogforwarder/validate_clusterlogforwarder_service_account_test.go

```diff
@@ -0,0 +1,152 @@
+package clusterlogforwarder
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	loggingv1 "github.com/openshift/cluster-logging-operator/apis/logging/v1"
+	"github.com/openshift/cluster-logging-operator/internal/constants"
+	authorizationapi "k8s.io/api/authorization/v1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+var _ = Describe("[internal][validations] validate clusterlogforwarder service account", func() {
+	var (
+		k8sClient client.Client
+		customClf loggingv1.ClusterLogForwarder
+		extras    map[string]bool
+	)
+
+	BeforeEach(func() {
+		k8sClient = fake.NewFakeClient() //nolint
+		customClf = loggingv1.ClusterLogForwarder{
+			ObjectMeta: v1.ObjectMeta{
+				Name:      "test-clf",
+				Namespace: constants.OpenshiftNS,
+			},
+		}
+		extras = map[string]bool{}
+	})
+
+	Context("service account existence", func() {
+		It("should fail when no service account found", func() {
+			serviceAccount, err := getServiceAccount(customClf.Spec.ServiceAccountName, customClf.Namespace, k8sClient)
+			Expect(serviceAccount).To(BeNil())
+			Expect(err).To(MatchError(MatchRegexp("service account: .+ not found")))
+		})
+
+		It("should succeed when service account is found", func() {
+			providedSAName := "test-serviceAccount"
+			clfServiceAccount := &corev1.ServiceAccount{
+				ObjectMeta: v1.ObjectMeta{
+					Name:      providedSAName,
+					Namespace: customClf.Namespace,
+				},
+			}
+			customClf.Spec.ServiceAccountName = providedSAName
+			Expect(k8sClient.Create(context.TODO(), clfServiceAccount)).To(Succeed())
```

Comment:
not necessarily needed to be change but FYI, you can use the builder for the fake client to add the object such that it will be returned without needing to "create it".  There are examples in the code base